### PR TITLE
core: Include a regular subsetted Noto Sans as the default font

### LINKED ISSUE #14761: minus sign does not appear
### Describe the bug

hi!
in a serie of questions for pupils, minus sign does never appear . Plus, divide or multiply are ok...
Here is a picture of it, and the swf zipped !
Thanks a lot
Franck

### Expected behavior

Show the sign minus, so that kids could choose right answer!

### Content Location

[exo307.zip](https://github.com/ruffle-rs/ruffle/files/13938003/exo307.zip)

### Affected platform

Browser's extension

### Operating system

Windows 11 

### Browser

Brave

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by evilpie**
The idea is to include a regular TTF font file for font rendering, instead of first creating a SWF with the Noto Sans as an embedded font. I am using `pyftsubset` together with a list of unicode ranges matching specific Unicode blocks.
This should make it quite easy to extend the included font in the future. Sadly `ttf_parser` doesn't support any of the newer compressed formats like WOFF. That is why I decided to manually apply deflate compression, which reduces the size by almost 50%.

Fixes #14761

### CODE REVIEW

File: tests/tests/swfs/avm2/stage3d_texture/output.expected.png
Comment (evilpie) [inline tests/tests/swfs/avm2/stage3d_texture/output.expected.png:1]: Not quite sure what happened here. I can't reproduce this when running the test in Firefox using the extension. (The colors seem wrong anyway)

image

File: tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png
Comment (evilpie) [inline tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png:1]: Oh this looks bad.

Comment (evilpie) [reply tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png:1]: Not sure what is going on here. The test actually includes a `DefineFont3` for `Noto Sans`, but it seems like we might still be falling back to the device font. I actually get the same rendering on desktop (I don't have `Noto Sans` installed).

Comment (evilpie) [reply tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png:1]: Apparently a known issues: https://github.com/ruffle-rs/ruffle/issues/14713. Thanks @kjarosh

**Comment by evilpie** [comment]
This should be ready to review. I don't see any text rendering change that seems problematic.

**Comment by torokati44** [comment]
> Sadly ttf_parser doesn't support any of the newer compressed formats like WOFF.

Just referencing https://github.com/RazrFalcon/ttf-parser/issues/122, in case it's added in the future.

**Comment by torokati44** [comment]
There are also crates like [allsorts](https://crates.io/crates/allsorts) and [webtype](https://crates.io/crates/webtype), have you considered relying on them (or anything else)? Not that there's anything wrong with simply deflating (it may be better even, as no new dependency brought in), I'm just curious.

**Comment by evilpie** [comment]
We need `ttf_parser` anyway for TLF's `DefineFont4`, so adding another crate with a different API would just be additional work.

File: tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png
Comment (kjarosh) [reply tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png:1]: This shouldn't be needed after https://github.com/ruffle-rs/ruffle/pull/18437.

File: tests/tests/swfs/avm2/stage3d_texture/output.expected.png
Comment (kjarosh) [reply tests/tests/swfs/avm2/stage3d_texture/output.expected.png:1]: What do you mean? Note that this test clicks the "Green Only" button (see its `input.json`).

Comment (evilpie) [reply tests/tests/swfs/avm2/stage3d_texture/output.expected.png:1]: Oh sorry. Somehow I thought it's obvious. But the = sign (e.g. in `rows=5`) now looks like just a dash (-).

Comment (kjarosh) [reply tests/tests/swfs/avm2/stage3d_texture/output.expected.png:1]: I believe you would have to render it in 1:1 scale and quality 1 to observe that. If we up the quality to 4 the second dash would probably be shown, but it's interesting it's missing on quality 1 :thinking:

**Review by Dinnerbone — APPROVED** [review]
Looks good to me but I'd really like to consider using metric compatible fonts at some point.

File: tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png
Comment (torokati44) [reply tests/tests/swfs/visual/cache_as_bitmap/edittext_selection/output.02.expected.png:1]: Dropped this change.

**Comment by evilpie** [comment]
> Looks good to me but I'd really like to consider using metric compatible fonts at some point.

Which is also a kind of fun point, because from what I can tell Flash on Linux never used the "standard" fonts from Windows either.

**Pull request merged**